### PR TITLE
Sparkle: update server url and enable background update when browser stars

### PR DESCRIPTION
**Pull request by simonhong**
Request explicit update check when about page loads
TODO: about page should show update status like windows version does.
issue: https://github.com/brave/brave-browser/issues/396

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: brave_init_settings.gni [hunk 1]
```diff
@@ -8,10 +8,12 @@ declare_args() {
...
 
-  # Update feed url on MacOS
-  # TODO(simonhong): This url is stub. Replace with real update server or your
-  # test server.
-  brave_feed_url = "http://localhost:9090"
-
   brave_dsa_file = "dsa_pub.pem"
 }
+
+_base_update_url = "https://updates-test.bravesoftware.com/sparkle/Brave-Browser"
```
Comment (simonhong) [inline brave_init_settings.gni]: @RyanJarv this url should be changed with real url

Comment (RyanJarv) [reply brave_init_settings.gni]: Yeah let's change the domain to updates.bravesoftware.com. (won't resolve now, but should be public in a bit)

Comment (RyanJarv) [reply brave_init_settings.gni]: Went ahead and updated that

**Comment by RyanJarv** [comment]
Added the update url we'll be using. Looks good to me, will wait for another set of eyes on this as well though.

File: brave_init_settings.gni [hunk 2]
```diff
@@ -8,10 +8,12 @@ declare_args() {
...
 
-  # Update feed url on MacOS
-  # TODO(simonhong): This url is stub. Replace with real update server or your
-  # test server.
-  brave_feed_url = "http://localhost:9090"
-
   brave_dsa_file = "dsa_pub.pem"
 }
+
+_base_update_url = "https://updates.bravesoftware.com/sparkle/Brave-Browser"
```
[the remark this answers is not in the document]
Comment (simonhong) [reply brave_init_settings.gni]: done.

**Pull request merged**